test/unit-tests/adapters/wormhole/WormholeReceiverAdapter.t.sol

```diff
@@ -0,0 +1,306 @@
+            receiverAdapter: address(43),
+            finalDestination: receiverAddr,
+            data: abi.encode(message)
+        });
+
+        adapter.receiveWormholeMessages(
+            abi.encode(payload), new bytes[](0), TypeCasts.addressToBytes32(senderAdapter), uint16(2), bytes32("1234")
+        );
+        vm.expectRevert(abi.encodeWithSelector(MessageExecutor.MessageIdAlreadyExecuted.selector, msgId));
+        adapter.receiveWormholeMessages(
+            abi.encode(payload), new bytes[](0), TypeCasts.addressToBytes32(senderAdapter), uint16(2), bytes32("1234")
+        );
+    }
+
+    /// @dev cannot receive message with invalid final destination
+    function test_receive_wormhole_messages_invalid_final_destination() public {
+        vm.startPrank(POLYGON_RELAYER);
+
+        address senderAdapter = contractAddress[SRC_CHAIN_ID]["WORMHOLE_SENDER_ADAPTER"];
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(42),
+            nonce: 0,
+            callData: bytes("42"),
+            nativeValue: 0,
+            expiration: type(uint256).max
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        AdapterPayload memory payload = AdapterPayload({
+            msgId: msgId,
+            senderAdapterCaller: address(42),
+            receiverAdapter: address(43),
+            finalDestination: address(44),
+            data: abi.encode(message)
+        });
+
+        vm.expectRevert(Error.INVALID_FINAL_DESTINATION.selector);
+        adapter.receiveWormholeMessages(
+            abi.encode(payload), new bytes[](0), TypeCasts.addressToBytes32(senderAdapter), uint16(2), bytes32("1234")
+        );
+    }
+
+    /// @dev reverts if message fails to be received
+    function test_receive_wormhole_messages_message_failure() public {
+        vm.startPrank(POLYGON_RELAYER);
+
+        address senderAdapter = contractAddress[SRC_CHAIN_ID]["WORMHOLE_SENDER_ADAPTER"];
+        address receiverAddr = contractAddress[DST_CHAIN_ID]["MMA_RECEIVER"];
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(0), // invalid target
+            nonce: 0,
+            callData: bytes("42"),
+            nativeValue: 0,
+            expiration: type(uint256).max
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        AdapterPayload memory payload = AdapterPayload({
+            msgId: msgId,
+            senderAdapterCaller: address(42),
+            receiverAdapter: address(43),
+            finalDestination: receiverAddr,
+            data: abi.encode(message)
+        });
+
+        // NOTE: Forge mangles low level error and doesn't allow checking for partial signature match
+        vm.expectRevert();
```

Comment:
suggest the following to test the correct error:
```suggestion
        vm.expectRevert(abi.encodeWithSelector(MessageExecutor.MessageFailure.selector, msgId, abi.encodePacked(Error.INVALID_TARGET.selector)));
```

Reply:
Hmm, for some reason that didn't work the first time I tried and I thought it was because Forge mangled it (See https://book.getfoundry.sh/cheatcodes/expect-revert#description). Working now.